### PR TITLE
WIP home page

### PR DESCRIPTION
**Pull request by diegodrf**
@linkdotnet I implemented the logic to manage the route flow as I mentioned.

What do you think about it? I know it can couple a little bit the project, but if the project is limited to only these pages I think it's not a problem.

It's a WIP, just to get your feedback.

### CODE REVIEW

File: LinqMarbles.Client/Components/Layout/MainLayout.razor [hunk 1]
```diff
@@ -9,7 +9,9 @@
         <MudHidden Breakpoint="Breakpoint.MdAndDown" Invert="true">
             <MudIconButton Icon="@Icons.Material.Filled.Menu" Edge="Edge.Start" Color="Color.Inherit" OnClick="@(() => _drawerOpen = !_drawerOpen)"/>
         </MudHidden>
-        <MudImage Src="favicon.png" Alt="logo" Width="30"/>
+        <MudLink Href="/">
```
Comment (linkdotnet) [inline LinqMarbles.Client/Components/Layout/MainLayout.razor:12]: Good catch!

File: LinqMarbles.Client/RouteConstants.cs [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        },
+        { "Set", [
+                RouteConstant.Except,
+                RouteConstant.Intersect,
+                RouteConstant.Union] 
+        }
+    }.ToFrozenDictionary();
+    
+
+    public static string GetRelativePath(this string path) => path.TrimStart('/');
```
Comment (linkdotnet) [inline LinqMarbles.Client/RouteConstants.cs:135]: This seems unused

File: LinqMarbles.Client/RouteConstants.cs [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    public static string GetPathTitle(this string path)
+    {
+        return string.Join(' ', path
+            .Split('/')
+            .Last()
+            .Split('-')
+            .Select(x => x[0].ToString().ToUpper() + x[1..]));
+    }
+
+    public static IEnumerable<string> GetAllRoutes()
```
Comment (linkdotnet) [inline LinqMarbles.Client/RouteConstants.cs:145]: Isn't it easier to use:
```csharp
return NavGroups.SelectMany(s => s.Value);
````
?

**Review by linkdotnet — COMMENTED** [review]
I like the general direction. Thanks for that @diegodrf 
Here are a few suggestions/things I noticed:
 - [ ] The operator text is not centered anymore
 - [ ] If I go to any "weird" route like "localhost:5000/test" it will show an empty page, as it is now considered an empty group
 - [ ] Search doesn't work

The second part keeps me thinking if we want to have the generic approach or, better have folders and dedicated pages per "group" to come around that fact. So, unknown urls aren't mapped and an empty page is shown. Of course we could also check, whether or not the list is empty, and if so, show some error message, but it seems a bit over the top.

**Comment by diegodrf** [comment]
Thank you for your comments.

Maybe I overcomplicated the code. I will discard it and rethink it.

**Comment by diegodrf** [comment]
I am closing the PR for now.

**Comment by linkdotnet** [comment]
> Thank you for your comments.
> 
> Maybe I overcomplicated the code. I will discard it and rethink it.

Nah don't worry - I like the general direction/idea.

**Pull request closed without merging**